### PR TITLE
return a string

### PR DESCRIPTION
**Pull request by qadan**
**JIRA Ticket**: https://jira.duraspace.org/browse/ISLANDORA-2006

# What does this Pull Request do?

Makes it so that `islandora_solr_truncate_field_display()` always returns a string instead of only returning a string if `$display_values` contains entries.

# What's new?
This came up as a result of trying to do whole-field truncation on an Islandora Solr Metadata display that may not necessarily have results. When this is the case, `islandora_solr_truncate_field_display()` takes the empty array and returns it instead of reformatting it into a string. It then tries to print the array, which breaks the display field.

This change adds an `else` in the case where the display field count is not `> 0` that ensures that a string is returned, albeit empty. The return string contains no markup, as the `<span>` statement for the truncated version is only required to facilitate the wrapper toggling.

# How should this be tested?
* Enable the Islandora Solr Metadata metadata display. Make sure in the Islandora Solr Metadata general confguration that  'Omit Empty Values' is not checked.
* Create an Islandora Solr Metadata display containing a field which may not have values.
* Set the field's Truncation Type to 'Limit Length of the whole field'; give it a Max Length greater than 0.
* Navigate to an object which doesn't implement this field.
* Explosions

# Additional Notes:
This change will break existing code relying on `islandora_solr_truncated_field_display()` returning an array. I cannot imagine that this is the case anywhere; the function claims to only return a string, so if anyone did that anywhere, well, tell e'm to read the docs I guess?

# Interested parties
@nhart maintains this, but this is a bug, so @Islandora/7-x-1-x-committers may wanna weigh in.

### CODE REVIEW

**Comment by DiegoPino** [comment]
@qadan i know this will sound picky and your fix is actually correct, but there is a coded logic that i don't like in that original piece that could have led to that original issue and maybe can be fixed here:

That phpdoc defines @return of type string. That is right.

At https://github.com/Islandora/islandora_solr_search/pull/321/files#diff-2d69368f59ba81d5635cd8274dd66090R489 we assign the first argument  of 
```PHP 
islandora_solr_truncate_field_display ($display_values, $max_length, $add_ellipsis, $word_safe, $wordsafe_length, $separator, $link_options = NULL) 
````
 we assign the first argument , which is in fact an array() to the variable we are planning to return, named `$updated_display_values`
That is in my opinion a logic issue or at least prone to lead to confusion. 

Would it not be easier (sorry, wrong, your solution is easier, but this could serve as an example for new coders!) to initialize $updated_display_values empty string, do all logic over the original  `$display_values`, assign strings to `$updated_display_values` during the logic if is needed or applies, .iIn that case, if no logic is needed, an empty string will be returned. 

The whole original, "i was an array" at the beginnig, and inbetween that HTML building logic, "i became a string", but not really, just in some cases,  is weird for my taste

**Comment by qadan** [comment]
@DiegoPino I agree with you here for sure - in fact, when I started trying to fix this, I took it from that tack. It's definitely weird and unnecessary. The part I kind of got stuck at is where we start `t()`-translating the resultant markup ... I'm not 100% familiar with the internals if `t()` so I really wanna make sure that this doesn't wipe out any existing site translations by virtue of ... I dunno, moving around the `t()`-references or changing what they contain.

I can change this to make more sense, it'll just take me a bit to make sure that translations are unaffected.

**Comment by DiegoPino** [comment]
@qadan true. HTML mixed with t() all over the place all as strings does not help much. A refactoring would require maybe some render arrays and a general t() for placeholder replacement. Not sure. I will approve this, for now, if you want to give it a try on a mayor refactor, I'm willing to follow up this and test and merge. If you don't have time then i will leave this approved and ask @Islandora/7-x-1-x-committers to merge as-is.

**Review by DiegoPino — APPROVED** [review]
Approving as it fixes the issue. Noting that i would prefer a refactor where the original argument is not passed as an array to string using the same variable name.

**Comment by jordandukart** [comment]
Bump @Islandora/7-x-1-x-committers

**Comment by whikloj** [comment]
👍  Always returns a string. Sorry to anyone using the current implementation instead of the PHPDocs. 😄

**Pull request merged**
